File: CalenderView/app/src/main/java/com/non_name_hero/calenderview/utils/DiskIOThreadExecutor.java
package com.non_name_hero.calenderview.utils;

import androidx.annotation.NonNull;

import java.util.concurrent.Executor;
import java.util.concurrent.Executors;

/**
 * バックグラウンドで処理するためのスレッドを管理するExecutorを実装するクラス
 *
 */
public class DiskIOThreadExecutor implements Executor {

    private final Executor mDiskIO;

    //コンストラクタ
    //一つのスレッドを管理するExecutorを返却
    public DiskIOThreadExecutor() {
        mDiskIO = Executors.newSingleThreadExecutor();
    }

    /**
     * 引数のRunnableをI/Oスレッドで実行する
     * @param command 実行したい処理内容
     */
    @Override
    public void execute(@NonNull Runnable command) {
        mDiskIO.execute(command);
    }
}

File: CalenderView/app/src/main/java/com/non_name_hero/calenderview/utils/DiskIOThreadExecutor.kt
package com.non_name_hero.calenderview.utils

import java.util.concurrent.Executor
import java.util.concurrent.Executors

/**
 * バックグラウンドで処理するためのスレッドを管理するExecutorを実装するクラス
 *
 */
class DiskIOThreadExecutor : Executor {
    private val mDiskIO: Executor

    /**
     * 引数のRunnableをI/Oスレッドで実行する
     * @param command 実行したい処理内容
     */
    override fun execute(command: Runnable) {
        mDiskIO.execute(command)
    }

    //コンストラクタ
    //一つのスレッドを管理するExecutorを返却
    init {
        mDiskIO = Executors.newSingleThreadExecutor()
    }
}